cache/src/test/java/io/envoyproxy/controlplane/cache/V2SimpleCacheTest.java

```diff
@@ -0,0 +1,546 @@
+package io.envoyproxy.controlplane.cache;
```

Comment:
With the tests, my approach was basically to rename the existing test to V2, and then copy/paste to a V3 class, updating the imports to use v3 message types, and using v3 type URLs. I also had to implement the new callbacks interfaces, and throw IllegalStateException in the test if the unexpected request version happens.

This means theres' a lot of copy/paste, but IMO that's less concerning for tests. I like tests to be clear rather than super DRY, whereas the implementation being reused as much as possible makes less surface for bugs.

I'm open to feedback on this though

Reply:
I'll do a second round and try to dry them up myself and see if that is better.